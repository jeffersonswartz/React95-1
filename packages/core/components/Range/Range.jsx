import styled, { css } from '@xstyled/styled-components';
import { th } from '@xstyled/system';

const trackStyle = css`
  width: 100%;
  height: 4px;

  background: ${th('colors.black')};

  cursor: pointer;

  border-top: 1;
  border-top-color: grays.1;
  border-right: 2;
  border-right-color: ${th('colors.white')};
  border-bottom: 1;
  border-bottom-color: grays.0;
  border-left: 2;
  border-left-color: ${th('colors.white')};

  box-shadow: 0px 1px 0px 0px ${th('colors.white')};
`;

const thumbStyle = css`
  width: 12px;
  height: 20px;

  background-color: bg;

  cursor: pointer;

  margin-top: -${th('space.8')}px;

  border-top: 1;
  border-top-color: ${th('colors.white')};
  border-right: 1;
  border-right-color: ${th('colors.black')};
  border-bottom: 1;
  border-bottom-color: ${th('colors.black')};
  border-left: 1;
  border-left-color: ${th('colors.white')};

  box-shadow: inset 0px -1px 0px ${th('colors.grays.3')},
    inset -1px 0px 0px ${th('colors.grays.3')},
    inset 0px 1px 0px ${th('colors.grays.0')},
    inset 1px 0px 0px ${th('colors.grays.0')};

  -webkit-appearance: none;
`;

const Range = styled.input.attrs({
  type: 'range',
})`
  -webkit-appearance: none;
  width: 100%;

  &:hover,
  &:focus,
  &:active {
    outline: none;
  }

  &::-webkit-slider-runnable-track {
    ${trackStyle}
  }

  &::-webkit-slider-thumb {
    ${thumbStyle}
  }

  &:focus::-webkit-slider-runnable-track {
    background: ${th('colors.black')};
  }

  &::-moz-range-track {
    ${trackStyle}
  }

  &::-moz-range-thumb {
    ${thumbStyle}
  }
`;

export default Range;
