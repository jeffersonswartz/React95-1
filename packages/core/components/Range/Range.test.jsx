import React from 'react';
import { render } from '../shared/test/utils';
import Range from './Range';

describe('<Range />', () => {
  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const { container } = render(<Range />);
      expect(container).toMatchSnapshot();
    });
  });
});
