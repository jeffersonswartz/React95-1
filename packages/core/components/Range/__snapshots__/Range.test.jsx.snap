// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Range /> Snapshot should match snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  width: 100%;
}

.c0:hover,
.c0:focus,
.c0:active {
  outline: none;
}

.c0::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  background: #000000;
  cursor: pointer;
  border-top: 1px solid;
  border-top-color: #d2d2d2;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 1px solid;
  border-bottom-color: #e6e6e6;
  border-left: 2px solid;
  border-left-color: #ffffff;
  box-shadow: 0px 1px 0px 0px #ffffff;
}

.c0::-webkit-slider-thumb {
  width: 12px;
  height: 20px;
  background-color: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid;
  border-top-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #000000;
  border-bottom: 1px solid;
  border-bottom-color: #000000;
  border-left: 1px solid;
  border-left-color: #ffffff;
  box-shadow: inset 0px -1px 0px #868a8e, inset -1px 0px 0px #868a8e, inset 0px 1px 0px #e6e6e6, inset 1px 0px 0px #e6e6e6;
  -webkit-appearance: none;
}

.c0:focus::-webkit-slider-runnable-track {
  background: #000000;
}

.c0::-moz-range-track {
  width: 100%;
  height: 4px;
  background: #000000;
  cursor: pointer;
  border-top: 1px solid;
  border-top-color: #d2d2d2;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 1px solid;
  border-bottom-color: #e6e6e6;
  border-left: 2px solid;
  border-left-color: #ffffff;
  box-shadow: 0px 1px 0px 0px #ffffff;
}

.c0::-moz-range-thumb {
  width: 12px;
  height: 20px;
  background-color: #c3c7cb;
  cursor: pointer;
  margin-top: -8px;
  border-top: 1px solid;
  border-top-color: #ffffff;
  border-right: 1px solid;
  border-right-color: #000000;
  border-bottom: 1px solid;
  border-bottom-color: #000000;
  border-left: 1px solid;
  border-left-color: #ffffff;
  box-shadow: inset 0px -1px 0px #868a8e, inset -1px 0px 0px #868a8e, inset 0px 1px 0px #e6e6e6, inset 1px 0px 0px #e6e6e6;
  -webkit-appearance: none;
}

<div>
  <input
    class="c0"
    type="range"
  />
</div>
`;
